### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/elsa/group/worker.ex [hunk 1]
```diff
@@ -77,12 +78,19 @@ defmodule Elsa.Group.Worker do
...
+      {:noreply, %{state | subscriber_pid: pid}}
+    else
       {:error, reason} ->
+        Logger.warn(
+          "Unable to subscribe to topic/partition/offset(#{state.topic}/#{state.partition}/#{state.offset}), reason #{
+            inspect(reason)
+          }"
+        )
+
+        Process.sleep(@start_failure_delay)
```
Comment (ManApart) [inline lib/elsa/group/worker.ex]: I'd be curious to here why we're sleeping here. I'm guessing we're waiting for something else to stop that we can't easily hook into?

File: lib/elsa/group/worker.ex [hunk 2]
```diff
@@ -136,8 +143,7 @@ defmodule Elsa.Group.Worker do
 
   defp subscribe(state, retries \\ @subscribe_retries)
 
-  defp subscribe(state, 0) do
-    Logger.error("Unable to subscribe to topic #{state.topic} partition #{state.partition} offset #{state.offset}")
+  defp subscribe(_state, 0) do
```
Comment (ManApart) [inline lib/elsa/group/worker.ex]: Why are we no longer erring here?

File: lib/elsa/producer/manager.ex [hunk 3]
```diff
@@ -0,0 +1,111 @@
+defmodule Elsa.Producer.Manager do
+  @moduledoc """
+  Define a to create and monitor producer processes
```
Comment (ManApart) [inline lib/elsa/producer/manager.ex]:
```suggestion
  Define a server to create and monitor producer processes
```

File: lib/elsa/group/worker.ex (hunk 1 above)
Comment (jeffgrunewald) [reply lib/elsa/group/worker.ex]: Sleeping forces the client to wait before exit so it doesn’t immediately trigger the supervisor to self-destruct due to the restart limit when there’s a network issue beyond our control.

Comment (ManApart) [reply lib/elsa/group/worker.ex]: Aaahhhh. JMS.

**Pull request merged**